src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -59,6 +62,18 @@ private BoundExpression RewriteCollectionExpressionConversion(Conversion convers
             }
         }
 
+        private BoundExpression VisitImmutableArrayCollectionExpression(BoundCollectionExpression node, TypeSymbol elementType)
+        {
+            var elementTypeWithAnnotations = TypeWithAnnotations.Create(elementType);
+            var arrayCreation = VisitArrayOrSpanCollectionExpression(
+                node,
+                CollectionExpressionTypeKind.Array,
+                ArrayTypeSymbol.CreateSZArray(_compilation.Assembly, elementTypeWithAnnotations),
+                elementTypeWithAnnotations);
+            var asImmutableArray = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_InteropServices_ImmutableCollectionsMarshal__AsImmutableArray_T)!;
+            return _factory.StaticCall(asImmutableArray.Construct(elementType), ImmutableArray.Create(arrayCreation));
```

Comment:
It doesn't look like this is checking things such as:
1. Is `AsImmutableArray` obsolete
2. Is `AsImmutableArray` `UnmanagedCallersOnly`

Reply:
Also, we should consider doing constraints checks on the method since we're calling `Construct`. May be worth discussing with the team how much we care about such custom BCL APIs...